src/Core/ViewHelper/Traits/TagGenerating.php

```diff
@@ -0,0 +1,70 @@
+<?php
+declare(strict_types=1);
+namespace TYPO3Fluid\Fluid\Core\ViewHelper\Traits;
+
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+use TYPO3Fluid\Fluid\Core\ViewHelper\ArgumentDefinition;
+use TYPO3Fluid\Fluid\Core\ViewHelper\TagBuilder;
+
+/**
+ * Tag Generating ViewHelper Trait
+ *
+ * Implement in ViewHelpers which should generate tags.
+ */
+trait TagGenerating
+{
+    /**
+     * @return ArgumentDefinition[]
+     */
+    abstract public function prepareArguments();
```

Comment:
Well, this is really hard-to-grasp for people not writing the code itself.
- trait has an abstract public method, but the concrete implementation does not implement it, but the abstractviewhelper implements this.
So this trait should be a an "abstract trait" de-facto. However, "abstract" is part of the AbstractViewHelper but not the TagViewHelper itself.

In addition:
- I don't see a benefit of introducing a TagGenerating trait which is used only in one place (TagViewHelper) which in turn overrides the only method - createTag() viable for TagViewHelper. Instead we could use the new TagViewHelper to extend for new tags directly.

Reply:
> trait has an abstract public method, but the concrete implementation does not implement it

That is the way we declare that the implementer must also implement that method if it wishes to implement the trait. Doesn't make the trait abstract-like ;)

> I don't see a benefit of introducing a TagGenerating trait which is used only in one place (TagViewHelper)

It is meant to replace the current method of tag capable ViewHelpers but cannot do so by changing the AbstractTagBasedViewHelper class because of 1) the differences in behavior which would be breaking and 2) the desire to make any VH capable of rendering tags rather than demanding the abstract class be implemented.

PS: if no usages of a feature is a reason for dropping, the AbstractTagBasedViewHelper would not exist because Fluid itself does not use it ;)